### DISCUSSION TITLE
Color schemes and wezterm-mux-server

### DISCUSSION THREAD

**Discussion by jason0x43** [wezterm/wezterm#2205] [Q&A] [answered]
I've been having trouble getting the wezterm theme to respond to my system's (macOS) current appearance. After a bit of experimentation, I believe the issue is that the config override process described on the [window:get_appearance() page](https://wezfurlong.org/wezterm/config/lua/window/get_appearance.html) doesn't work for windows attached to `wezterm-mux-server`.

If I start a GUI instance with `wezterm connect unix` (my config creates a unix domain named "unix"), the window will only update its theme if I edit and save the `wezterm.lua` file. If my config sets a `color_scheme` config override in response to a "window-config-reloaded" event, GUI windows won't update to use the new theme.

If I start a GUI instance _without_ connecting to a mux server, an override theme _will_ work. I can set `color_scheme` in an override object and call `window:set_config_overrides(overrides)`, and the window theme will update.

Is this expected behavior? I read through #1268, which seemed like it might be relevant, but I'm not using a remote system, so I wasn't sure.

**Comment by wez** [comment, marked as answer]
The mux server is "remote" from the local GUI process wrt. config overrides: the GUI-window-specific overrides only apply to the GUI process and are not visible to the "remote" mux.

I'd recommend not using a local unix mux server for most users; wezterm actually starts an embedded mux inside the gui process anyway, and `wezterm cli` knows how to resolve it and communicate with it. Cutting out a separate mux process can make a big difference to performance as well.

I'd suggest only running a separate mux server if you plan to connect to that machine from a separate machine; eg: that machine is sitting in your office and you're connecting from home.

**Reply by jason0x43** [reply]
Thanks, that makes sense. The local mux server reminded me of using tmux, where the core terminal and the UI windows are separate things, which I always thought was pretty nice (particularly when I accidentally closed the terminal app 😄).
